internal/dlog/dlog_test.go

```diff
@@ -30,19 +31,21 @@ type params struct {
 	p, order, g *big.Int
 }
 
-func get_params() *params {
-	p, _ := emmy.GetSafePrime(20)
-	pMin1 := new(big.Int).Sub(p, big.NewInt(1))
-	x := emmy.GetRandomInt(p)
+func get_params(t *testing.T) *params {
+	key, err := keygen.NewElGamal(20)
+	if err != nil {
+		t.Fatalf("Error during parameters generation: %v", err)
+	}
+	pMin1 := new(big.Int).Sub(key.P, big.NewInt(1))
```

Comment:
You can avoid declaring a variable and  move `new(big.Int).Sub(key.P, big.NewInt(1))` to L42